code/game/turfs/simulated/wall_attacks.dm

```diff
@@ -71,6 +71,17 @@
 			fail_smash(user, 2)
 			return 1
 
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/turf/destination = GetAbove(H)
+
+		if(destination)
+			var/turf/start = get_turf(H)
+			if(start.CanZPass(H, UP))

```

Comment:
Lessen the doom a bit and do `if (start.CanZPass(H, UP) && destination.CanZPass(H, UP))`